src/test/java/picard/sam/AddOATagTest.java

```diff
@@ -0,0 +1,112 @@
+                new Object[]{new File("testdata/picard/sam/aligned.sam"), new File("testdata/picard/sam/contiguous.interval_list")}
+        };
+    }
+
+    @DataProvider(name="overWriteTestData")
+    Object[][] overwriteTestData() {
+        return new Object[][]{
+                new Object[]{new File("testdata/picard/sam/aligned.sam"), null, true},
+                new Object[]{new File("testdata/picard/sam/aligned.sam"), null, false}
+        };
+    }
+
+    @Test(dataProvider = "testData")
+    public void testTagWriting(final File input, final File interval_list) throws IOException {
+        final File clpOutput = File.createTempFile("AddOATag", ".bam");
+        clpOutput.deleteOnExit();
+
+        runAddOATag(input, clpOutput, interval_list, null);
+
+        validateBamForOA(clpOutput, interval_list, null);
+    }
+
+    @Test(dataProvider = "overWriteTestData")
+    public void testOverWrite(final File input, final File interval_list, Boolean overWriteTag) throws IOException {
+        final File firstPassOutput = File.createTempFile("FirstPassAddOATag", ".bam");
+        firstPassOutput.deleteOnExit();
+        final File secondPassOutput = File.createTempFile("SecondPassAddOATag", ".bam");
+        secondPassOutput.deleteOnExit();
+
+        // make first pass bam that only has one value per OA tag
+        runAddOATag(input, firstPassOutput, interval_list, true);
+
+        // make bam we want to test the overwrite option with
+        runAddOATag(firstPassOutput, secondPassOutput, interval_list, overWriteTag);
+
+        validateBamForOA(secondPassOutput, interval_list, overWriteTag);
+    }
+
+    private void runAddOATag(final File input, final File output, final File intervalList, Boolean overwriteTag) throws IOException {
+        final ArrayList<String> args = new ArrayList<String>(){
+            {
+                add("INPUT=" + input);
+                add("OUTPUT=" + output);
+            }
+        };
+        if (intervalList != null) {
+            args.add("INTERVAL_LIST=" + intervalList);
+        }
+        if (overwriteTag != null) {
+            args.add("OVERWRITE_TAG=" + overwriteTag);
+        }
+        AddOATag addOATag = new AddOATag();
+        Assert.assertEquals(addOATag.instanceMain(args.toArray(new String[args.size()])), 0, "Running addOATag did not succeed");
+    }
+
+    private void validateBamForOA(final File input, final File intervalList, Boolean overwriteTag) {
+        final SamReader reader = SamReaderFactory.makeDefault().open(input);
+        final SAMRecordIterator iterator = reader.iterator();
+
+        OverlapDetector<Interval> detector = AddOATag.getOverlapDetectorFromIntervalListFile(intervalList, 0, 0);
+        while (iterator.hasNext()){
+            SAMRecord rec = iterator.next();
+            if (!rec.getReadUnmappedFlag() && (detector == null || detector.overlapsAny(rec))){
+                Assert.assertNotNull(rec.getAttribute("OA"));
```

Comment:
Non-null does not mean that is correct, just that it was set; I prefer the file-based test due to that.